src/SourceBuild/content/repo-projects/package-source-build.proj

```diff
@@ -9,61 +9,58 @@
-    <Copy SourceFiles="$(CurrentSourceBuiltPackageVersionPropsPath)"
-          DestinationFiles="$(SourceBuildTarballStagingDir)PackageVersions.props" />
-
-    <ItemGroup>
-      <SourceBuildReferencePackagesNupkgFiles Include="$(ReferencePackagesDir)**/*.nupkg" />
-    </ItemGroup>
-
-    <!-- Copy reference packages from ReferencePackagesDir to staging dir. -->
-    <Copy
-      SourceFiles="@(SourceBuildReferencePackagesNupkgFiles)"
-      DestinationFiles="@(SourceBuildReferencePackagesNupkgFiles -> '$(SourceBuildReferencePackagesDestination)%(Filename)%(Extension)')"
-      Condition="'@(SourceBuildReferencePackagesNupkgFiles)' != ''" />
-
-    <ItemGroup>
-      <ArtifactsPackage Include="$(ArtifactsPackagesDir)**\*.nupkg" />
-    </ItemGroup>
-
-    <!-- Copy packages to staging dir. -->
-    <Copy SourceFiles="@(ArtifactsPackage)"
-          DestinationFolder="$(SourceBuildTarballStagingDir)"
-          Condition="'@(ArtifactsPackage)' != ''" />
-
     <PropertyGroup>
       <SourceBuiltTarballName>$(ArtifactsAssetsDir)$(SourceBuiltArtifactsTarballName).$(SourceBuiltSdkVersion).$(TargetRid)$(ArchiveExtension)</SourceBuiltTarballName>
       <SourceBuiltVersionFileName>.version</SourceBuiltVersionFileName>
     </PropertyGroup>
 
-    <!-- Content of the .version file to include in the tarball. Write to staging dir. -->
+    <!-- Content of the .version file to include in the tarball -->
     <ItemGroup>
       <VersionFileContent Include="$(RepositoryCommit);$(SourceBuiltSdkVersion)" />
     </ItemGroup>
 
     <WriteLinesToFile
-      File="$(SourceBuildTarballStagingDir)$(SourceBuiltVersionFileName)"
+      File="$(BaseIntermediateOutputPath)$(SourceBuiltVersionFileName)"
       Lines="@(VersionFileContent)"
       Overwrite="true" />
 
-    <MakeDir Directories="$([System.IO.Path]::GetDirectoryName('$(SourceBuiltTarballName)'))" />
-    <Exec Command="tar --numeric-owner -czf $(SourceBuiltTarballName) $(SourceBuiltVersionFileName) *.nupkg *.props SourceBuildReferencePackages/"
-          WorkingDirectory="$(SourceBuildTarballStagingDir)" />
+    <ItemGroup>
+      <_AllRepoDirs Include="$([System.IO.Directory]::GetDirectories($(ArtifactsShippingPackagesDir)))" />
+      <_AllRepoDirs Include="$([System.IO.Directory]::GetDirectories($(ArtifactsNonShippingPackagesDir)))" />
+      <_AllRepoFiles Include="%(_AllRepoDirs.Identity)/**" />
+      <_ReferencePackageFiles Include="$(ReferencePackagesDir)**" />
+    </ItemGroup>
 
-    <Message Importance="High" Text="Packaged source-built artifacts to $(SourceBuiltTarballName)" />
+    <!-- Create a layout directory for the files that are to be included in the artifacts tarball. Since there are a large number of files, this will use symlinks
+         instead of copying files to make this execute quickly. -->
+    <PropertyGroup>
+      <_SourceBuiltLayoutDir>$(BaseIntermediateOutputPath)artifacts-layout/</_SourceBuiltLayoutDir>
```

Comment:
Use the NormalizeDirectory msbuild function here. This makes the path look correct when looking at it on a binlog or when printing it to the console. While this project only runs on Unix today, it's generally a good pattern to follow.

Reply:
Fixed in https://github.com/dotnet/installer/pull/18557/commits/80ba49348ba3fcb28921d8f8fd0165a34838b59a